### ISSUE TITLE
Shoko generates different 

### ISSUE THREAD

**Issue by t0815** [ShokoAnime/ShokoServer#1092]
## VERSION INFORMATION

Server Version:  4.2.2

Desktop Version:  4.2.1

## LOG FILE

```
2023-10-24 18:15:12
Info
Get AniDB file info: 70542
2023-10-24 18:15:14
Info
File 70542 (96E58FE1043EFC6A3302FFC6FC620651 | [Vivid] Suisei no Gargantia - Meguru Kouro, Haruka OVA - 01 [BD 1080p FLAC] [3D363369].mkv) could not be found on AniDB
2023-10-24 18:15:14
Info
Get AniDB file info: 70543
2023-10-24 18:15:16
Info
File 70543 (F3104CE90318AF633A31C74C48C0AA5B | [Vivid] Suisei no Gargantia - Meguru Kouro, Haruka OVA - 02 [BD 1080p FLAC] [EDE34F81].mkv) could not be found on AniDB
2023-10-24 18:15:16
Info
Get AniDB file info: 70535
```

## DESCRIPTION

Very strange issue. Did add OVA for [Suisei no Gargantia: Meguru Kouro, Haruka](https://anidb.net/anime/10549 ) from SubGroup Vivid [EP 1 Nya](https://nyaa.si/view/636474), for the following files:

[Suisei no Gargantia: Meguru Kouro, Haruka - 1](https://anidb.net/file/1522368)
[Suisei no Gargantia: Meguru Kouro, Haruka - 1](https://anidb.net/file/1635377)

For some reason, shoko is generating a different ed2k hash than avdump for those files, while all other hashes do match:

EP1, Shoko Hashes:
**CRC 32:** 3D363369
**MD5**: F7725C6EFD96F65971A2EBFBE359E638
**SHA1**: 9012ED7854BFCC9AB9A11B8E3E4F1074D2A72CCE
**ed2k**: ed2k://|file|[Vivid] Suisei no Gargantia - Meguru Kouro, Haruka OVA - 01 [BD 1080p FLAC] [3D363369].mkv|4353287981|96E58FE1043EFC6A3302FFC6FC620651|/

AniDB File Info:

![grafik](https://github.com/ShokoAnime/ShokoServer/assets/10379828/112ba313-26a3-4832-8763-aed27670b806)

```
avdump result: ed2k://|file|[Vivid] Suisei no Gargantia - Meguru Kouro, Haruka OVA - 01 [BD 1080p FLAC] [3D363369].mkv|4353287981|A375B63AD1DA809BBE3A4AE31E399261|/

shoko result: ed2k://|file|[Vivid] Suisei no Gargantia - Meguru Kouro, Haruka OVA - 01 [BD 1080p FLAC] [3D363369].mkv|4353287981|96E58FE1043EFC6A3302FFC6FC620651|/
```

When i try to add the files to anidb, after dumping them i get error, files exist:

![grafik](https://github.com/ShokoAnime/ShokoServer/assets/10379828/b103fd45-2a51-4768-ac1c-fe277673095a)

Im not quite sure, what is going on here, why shoko for those files is generating a different ed2k hash. Other files are still importing fine so far.

## STEPS TO REPRODUCE
1.) download specified files
2,) import files
3.) compare shoko ed2k hash with anidb, avdump hash.

**Comment by harshithmohan** [comment]
Removed nyaa link

**Comment by revam** [comment]
Just to be sure, but you've tried rehashing the file(s), right? @t0815 

(Also, your Shoko Desktop is out of date, update it)

**Comment by t0815** [comment]
yep. but, its happening for other files now too lol wth??? I'm starting to think my shoko installation might have a small corruption. Never happened before, did not update or change something. CRCs and other hashes match... so makes no sense at all. I will reinstall shoko.

**Comment by t0815** [comment]
ok reinstall, no change... hmmm. I will look for an ed2k hasher tool lol and will check against calculated hash.... question is, who is wrong 😄

**Comment by t0815** [comment]
meh, well looks like shoko actually generates wrong hashes sometimes. Did rehash the files with eMule LinkCreator and got same result as AVDump.

**Comment by revam** [comment]
And did a rehash of the file in shoko fix the hash? Or is it still wrong even after ypu have performed a rehash? (I'm asking again since i didn't get a concrete answer "yes it fixed it" or "no it didn't fix it" yesterday) @t0815

**Comment by t0815** [comment]
oh sorry, rehash did not fix it! I deleted (renamed and deleted from shoko, since rehash isnt working currently) the files and imported them again, still wrong hash.

**Comment by revam** [comment]
Thank you for the clarifications.

@da3dsoul any ideas as to why the hash might be wrong a d rehashing is nit working (before i try to look into this)?
